Add `keys update` command to CLI

msabramo opened the pull request:
## Title

Add `keys update` command to CLI

```shell
$ litellm-proxy keys update \
  --key=2d7fd3dcfa6fb7a5142fcf234fc3e2e9dc983ad34ba435824929e8bb5ab28ebc \
  --key-alias=my-key \
  --user-id=me@company.com
{
  "key": "2d7fd3dcfa6fb7a5142fcf234fc3e2e9dc983ad34ba435824929e8bb5ab28ebc",
   ...
  "key_alias": "my-key",
   ...
  "user_id": "me@company.com",
   ...
}
```

## Relevant issues

<!-- e.g. "Fixes #000" -->

## Pre-Submission checklist

**Please complete all items before asking a LiteLLM maintainer to review your PR**

- [ ] I have Added testing in the [`tests/litellm/`](https://github.com/BerriAI/litellm/tree/main/tests/litellm) directory, **Adding at least 1 test is a hard requirement** - [see details](https://docs.litellm.ai/docs/extras/contributing_code)
- [ ] I have added a screenshot of my new test passing locally 
- [ ] My PR passes all unit tests on [`make test-unit`](https://docs.litellm.ai/docs/extras/contributing_code)
- [ ] My PR's scope is as isolated as possible, it only solves 1 specific problem


## Type

<!-- Select the type of Pull Request -->
<!-- Keep only the necessary ones -->

🆕 New Feature

## Changes

msabramo committed:
Add `keys update` command to CLI

msabramo committed:
keys.py formatting

ishaan-jaff commented:
@msabramo is this complete ?

msabramo commented:
> @msabramo is this complete ?

Yes I believe so. Give it a try and see if it does what you think it should do.

krrishdholakia commented on litellm/proxy/client/cli/commands/keys.py:
```diff
@@ -156,3 +156,130 @@ def delete(ctx: click.Context, keys: Optional[str], key_aliases: Optional[str]):
+@click.option("--temp-budget-expiry", type=str, help="Expiry time for the temporary budget increase")
+@click.option("--allowed-routes", type=str, help="Comma-separated list of allowed routes for the key")
+@click.pass_context
+def update(
```
Can we add a test / something to keep this in sync with the `/key/update` endpoint? 

Here's how we enforce the docstring for `/key/update` is always up-to-date - https://github.com/BerriAI/litellm/blob/3a782856ea85fe4c84a63496ddc35d1f05bb7e4c/tests/documentation_tests/test_api_docs.py#L119

msabramo replied:
> Can we add a test / something to keep this in sync with the `/key/update` endpoint?
> 
> Here's how we enforce the docstring for `/key/update` is always up-to-date -
> 
> https://github.com/BerriAI/litellm/blob/3a782856ea85fe4c84a63496ddc35d1f05bb7e4c/tests/documentation_tests/test_api_docs.py#L119

Check out the latest:

```
abramowi at Marcs-MBP-3 in ~/Code/OpenSource/litellm (proxy-client-cli-keys-update●)
$ ls -l tests/litellm/proxy/client/cli/test_cli_sync.py
-rw-r--r--  1 abramowi  staff  1637 May 13 16:46 tests/litellm/proxy/client/cli/test_cli_sync.py
```

krrishdholakia reviewed (changes requested):
add test to keep this in sync with `/key/update`

msabramo committed:
Add CLI sync test

addresses
[comment](https://github.com/BerriAI/litellm/pull/10599#discussion_r2083225447)
from @krrishdholakia

msabramo committed:
Remove unused soft_budget & send_invite_email params

that `test_cli_sync.py` detected!

```
E       AssertionError: Extra fields in CLI update command: {'soft_budget', 'send_invite_email'}
E       assert not {'send_invite_email', 'soft_budget'}

tests/litellm/proxy/client/cli/test_cli_sync.py:44: AssertionError
```

msabramo commented:
> add test to keep this in sync with `/key/update`

Check out the latest:

```
abramowi at Marcs-MBP-3 in ~/Code/OpenSource/litellm (proxy-client-cli-keys-update●)
$ ls -l tests/litellm/proxy/client/cli/test_cli_sync.py
-rw-r--r--  1 abramowi  staff  1637 May 13 16:46 tests/litellm/proxy/client/cli/test_cli_sync.py
```

msabramo commented:
Ready for rereview @krrishdholakia
